plugins/chat/assets/javascripts/discourse/components/chat/routes/channel-info-settings.gjs

```diff
@@ -98,10 +98,14 @@ export default class ChatRouteChannelInfoSettings extends Component {
     return this.args.channel.isCategoryChannel && this.args.channel.isOpen;
   }
 
-  get toggleThreadingAvailable() {
+  get toggleThreadingCategoryChannel() {
     return this.args.channel.isCategoryChannel && this.args.channel.isOpen;
   }
 
+  get toggleThreadingDirectMessage() {
+    return this.args.channel.isDirectMessageChannel && this.args.channel.isOpen;
+  }
```

Comment:
Since both types of channels now support threading, do we need to duplicate these methods? Wouldn't this just work?

```
get canToggleThreading() {
  return this.args.channel.isOpen;
}
```

Reply:
At the moment it's more of a permission issue, the public channels appear under the admin section and should only be updated by staff users, however for DMs we want all members of the channel to be able to toggle threads. This is the main reason for handling these separately.

I think if we decide to update the channel settings layout in future then this can be cleaned up.